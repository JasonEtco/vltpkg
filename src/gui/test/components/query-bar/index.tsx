import { vi, test, assert, expect, afterEach } from 'vitest'
import {
  cleanup,
  render,
  fireEvent,
  screen,
} from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.js'
import { QueryBar } from '@/components/query-bar/index.jsx'

vi.mock('@/components/query-bar/query-highlighter.jsx', () => ({
  QueryHighlighter: 'gui-query-highlighter',
}))

vi.mock('lucide-react', () => ({
  Search: 'gui-search',
  Command: 'gui-command',
}))

vi.mock('@/components/ui/kbd.jsx', () => ({
  Kbd: 'gui-kbd',
}))

vi.mock('@/components/explorer-grid/query-matches.jsx', () => ({
  QueryMatches: 'gui-query-matches',
}))

vi.mock('@/components/explorer-grid/save-query.jsx', () => ({
  default: 'gui-save-query',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

test('query-bar render default', async () => {
  render(<QueryBar />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('query-bar render with custom query', async () => {
  const Container = () => {
    const updateQuery = useStore(state => state.updateQuery)
    updateQuery('[name="my-package"][version="1.0.0"]')
    return <QueryBar />
  }
  render(<Container />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('query-bar updates query store value', async () => {
  render(<QueryBar />)

  const value = '[name="my-package"][version="2.0.0"]'
  fireEvent.change(screen.getByRole('search'), {
    target: { value },
  })

  let query = ''
  const RetrieveQuery = () => (
    (query = useStore(state => state.query)), ''
  )
  render(<RetrieveQuery />)

  assert.equal(query, '[name="my-package"][version="2.0.0"]')
})
