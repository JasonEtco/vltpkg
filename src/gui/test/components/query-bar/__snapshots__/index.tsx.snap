// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`query-bar render default 1`] = `

<div>
  <div class="relative flex grow items-center">
    <div class="absolute pr-8">
      <gui-search
        size="20"
        classname="ml-3 text-neutral-500"
      >
      </gui-search>
    </div>
    <div class="relative w-full">
      <input
        type="text"
        class="clear-none flex h-10 rounded-md border border-input px-3 py-2 ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 relative w-full bg-white pl-10 text-sm text-transparent caret-black dark:bg-muted-foreground/5 dark:caret-white"
        role="search"
        autocorrect="off"
        autocomplete="off"
        autocapitalize="off"
        placeholder="Query Lookup, e.g: :root > *"
        value=":root"
      >
      <gui-query-highlighter>
      </gui-query-highlighter>
    </div>
    <div class="absolute right-0">
      <div class="mr-3 hidden items-center gap-1 backdrop-blur-sm md:flex">
        <gui-query-matches>
        </gui-query-matches>
        <gui-save-query>
        </gui-save-query>
        <gui-kbd classname="before:content-[&quot; &quot;] relative ml-3 before:absolute before:-ml-10 before:h-[0.75rem] before:w-[1.25px] before:rounded-sm before:bg-neutral-600">
          <gui-command size="12">
          </gui-command>
        </gui-kbd>
        <gui-kbd classname="text-sm">
          k
        </gui-kbd>
      </div>
    </div>
  </div>
</div>

`;

exports[`query-bar render with custom query 1`] = `

<div>
  <div class="relative flex grow items-center">
    <div class="absolute pr-8">
      <gui-search
        size="20"
        classname="ml-3 text-neutral-500"
      >
      </gui-search>
    </div>
    <div class="relative w-full">
      <input
        type="text"
        class="clear-none flex h-10 rounded-md border border-input px-3 py-2 ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 relative w-full bg-white pl-10 text-sm text-transparent caret-black dark:bg-muted-foreground/5 dark:caret-white"
        role="search"
        autocorrect="off"
        autocomplete="off"
        autocapitalize="off"
        placeholder="Query Lookup, e.g: :root > *"
        value="[name=&quot;my-package&quot;][version=&quot;1.0.0&quot;]"
      >
      <gui-query-highlighter>
      </gui-query-highlighter>
    </div>
    <div class="absolute right-0">
      <div class="mr-3 hidden items-center gap-1 backdrop-blur-sm md:flex">
        <gui-query-matches>
        </gui-query-matches>
        <gui-save-query>
        </gui-save-query>
        <gui-kbd classname="before:content-[&quot; &quot;] relative ml-3 before:absolute before:-ml-10 before:h-[0.75rem] before:w-[1.25px] before:rounded-sm before:bg-neutral-600">
          <gui-command size="12">
          </gui-command>
        </gui-kbd>
        <gui-kbd classname="text-sm">
          k
        </gui-kbd>
      </div>
    </div>
  </div>
</div>

`;
